staging/src/k8s.io/api/storage/v1beta1/types.go

```diff
@@ -209,3 +209,174 @@ type VolumeError struct {
 	// +optional
 	Message string `json:"message,omitempty" protobuf:"bytes,2,opt,name=message"`
 }
+
+// VolumePluginMechanism is the type of mechanism components should use for volume operations
+type VolumePluginMechanism string
+
+const (
+	// VolumePluginMechanismInTree means components should use the in-tree plugin for volume operations
+	VolumePluginMechanismInTree VolumePluginMechanism = "in-tree"
+	// VolumePluginMechanismCSI means components should use the CSI Driver for volume operations
+	VolumePluginMechanismCSI VolumePluginMechanism = "csi"
+)
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CSIDriver captures information about a Container Storage Interface (CSI)
+// volume driver deployed on the cluster.
+// CSIDriver objects are non-namespaced.
+type CSIDriver struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard object metadata.
+	// metadata.Name indicates the name of the CSI driver that this object
+	// refers to; it MUST be the same name returned by the CSI GetPluginName()
+	// call for that driver.
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// Specification of the CSI Driver.
+	Spec CSIDriverSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CSIDriverList is a collection of CSIDriver objects.
+type CSIDriverList struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// Standard list metadata
+	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#metadata
+	// +optional
+	metav1.ListMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	// items is the list of CSIDriver
+	Items []CSIDriver `json:"items" protobuf:"bytes,2,rep,name=items"`
+}
+
+// CSIDriverSpec is the specification of a CSIDriver.
+type CSIDriverSpec struct {
+	// attachRequired indicates this CSI volume driver requires an attach
+	// operation (because it implements the CSI ControllerPublishVolume()
+	// method), and that Kubernetes should call attach and wait for any attach
+	// operation to complete before proceeding to mounting.
+	// If value is not specified, default is false -- meaning attach will not be
+	// called.
+	// +optional
+	AttachRequired bool `json:"attachRequired,omitempty" protobuf:"varint,1,opt,name=attachRequired"`
```

Comment:
I think the comments about the default value is not right here. If the CSIDriverRegistry feature gate is enabled and the value is specified to false, the attach operation will be skipped. Otherwise the attach operation will be called.  I'll change this back to a pointer and reword the comments to clarify.